app/src/main/java/org/openobservatory/ooniprobe/activity/oonirun/OoniRunV2Activity.kt

```diff
@@ -0,0 +1,217 @@
+package org.openobservatory.ooniprobe.activity.oonirun
+
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import android.widget.Toast
+import org.openobservatory.ooniprobe.R
+import org.openobservatory.ooniprobe.activity.AbstractActivity
+import org.openobservatory.ooniprobe.activity.adddescriptor.AddDescriptorActivity
+import org.openobservatory.ooniprobe.common.TaskExecutor
+import org.openobservatory.ooniprobe.common.TestDescriptorManager
+import org.openobservatory.ooniprobe.common.ThirdPartyServices
+import org.openobservatory.ooniprobe.databinding.ActivityOoniRunV2Binding
+import org.openobservatory.ooniprobe.model.database.TestDescriptor
+import javax.inject.Inject
+
+/**
+ * Activity to handle a v2 link.
+ *
+ * A v2 link has the following format:
+ *
+ * 1. ooni://runv2/link_id
+ * 2. https://run.test.ooni.org/v2/link_id
```

Comment:
This has not been discussed int the past. the final resting domain for OONI Run V2 has not been discussed. I am not sure at the moment what it would be changed to in the future.